test/logisticalTests.js

```diff
@@ -85,32 +86,144 @@ describe('Logistical', function(){
+         *
+         * Z[0] = 1*1
+         * logistic[0] = 1/(1 + exp(-1*1))
+         * sum[0] = log(1 * 1/(1 + exp(-1*1)))
+         *
+         * Z[1] = 1*3
+         * logistic[1] = 1/(1 + exp(-3*0))
+         * sum[1] = log(1/(1 + exp(-3*0)))
+         *
+         * Z[2] = 1*6
+         * logistic[2] = 1/(1 + exp(-1*6))
+         * sum[2] = log(1/(1 + exp(-1*6)))
+         *
+         * L(w) = sum[0] + sum[1] + sum[2]
+         */
+        // from wolfrom alpha
+        // http://www.wolframalpha.com/input/?i=log%281+*+1%2F%281+%2B+exp%28-1*1%29%29%29+%2B+log%281%2F%281+%2B+exp%28-3*0%29%29%29+%2B+log%281%2F%281+%2B+exp%28-1*6%29%29%29
+        assert.equal(-1.00888, subject.logLikelihood(w, Y, X, C).toFixed(5));
+      });
+
+      it('computes the logLikelihood for 2 dependent variables', function() {
+        var w = linear.Vector.create([1,1]);
+        var X = new linear.Matrix.create([[1,2],[3,4],[5,6]]);
+        var Y = [1,0,1];
+        // from wolfrom alpha
+        // http://www.wolframalpha.com/input/?i=log%28logistic+function+3%29+%2B+log%28logistic+function+0%29+%2B+log%28logistic+function+11%29
+        assert.equal(-0.74175, subject.logLikelihood(w, Y, X, C).toFixed(5));
+      });
+    });
+    describe('with regularization', function() {
+      var C = 0.1;
+
+      it('computes the logLikelihood for a single dependent variable', function() {
+        var w = linear.Vector.create([1]);
+        var X = new linear.Matrix.create([[1],[3],[6]]);
+        var Y = [1,0,1];
+
+        /*
+         * Simple enough to outline the component steps
+         *
+         * Z[0] = 1*1
+         * logistic[0] = 1/(1 + exp(-1*1))
+         * sum[0] = log(1 * 1/(1 + exp(-1*1)))
+         *
+         * Z[1] = 1*3
+         * logistic[1] = 1/(1 + exp(-3*0))
+         * sum[1] = log(1/(1 + exp(-3*0)))
+         *
+         * Z[2] = 1*6
+         * logistic[2] = 1/(1 + exp(-1*6))
+         * sum[2] = log(1/(1 + exp(-1*6)))
+         *
+         * L(w) = -(sum[0] + sum[1] + sum[2]) + 0.5 * C * w dot w
+         */
+        // from wolfrom alpha
+        // http://www.wolframalpha.com/input/?i=log%281+*+1%2F%281+%2B+exp%28-1*1%29%29%29+%2B+log%281%2F%281+%2B+exp%28-3*0%29%29%29+%2B+log%281%2F%281+%2B+exp%28-1*6%29%29%29
+        assert.equal(1.05888, subject.logLikelihood(w, Y, X, C).toFixed(5));
+      });
+
+      it('computes the logLikelihood for 2 dependent variables', function() {
+        var w = linear.Vector.create([1,1]);
+        var X = new linear.Matrix.create([[1,2],[3,4],[5,6]]);
+        var Y = [1,0,1];
+        // from wolfrom alpha
+        // http://www.wolframalpha.com/input/?i=log%28logistic+function+3%29+%2B+log%28logistic+function+0%29+%2B+log%28logistic+function+11%29
+        assert.equal(0.84175, subject.logLikelihood(w, Y, X, C).toFixed(5));
+      });
+    });
+  });
+
+  /* Test classification of a value based on some training data */
+  describe.skip('#classify', function() {
+    var i;
```

Comment:
While this seems clearer, Node.js has a very elaborate understanding of scope (http://speakingjs.com/es5/ch16.html). I find that getting into the practice of declaring var variables within new scopes saves you headaches later where you end up with scope spillover inadvertently. 

This is especially true with async code.